Java/core/src/main/java/com/amazon/randomcutforest/tree/CompactLeaf.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.tree;
+
+public class CompactLeaf {
+
+    private final int pointIndex;
+    private short parentIndex;
```

Comment:
question. why is one index int and the other short?

Reply:
The PointStore uses int indexes, because the maximum number of points that we need to store (number of trees * sample size) is on the same order as the maximum short value (~32,000). We thought that using a short index for points could could unnecessarily restrict valid use cases.

Reply:
is the short parent index already a restriction? might as well make them consistent.

Reply:
The reason for the discrepancy is that the current design uses one point store for the forest, but it uses one node store and one leaf store for each tree in the forest.